### PR TITLE
Bump ControllerGen, CRD 

### PR DESCRIPTION
**Pull request by kuritka**
I'm upgrading the kube-linter to the latest version. I need to do this before I bump the depenencies in the go.mod files.  This change modifies the CRD! 

..

<details>
  <summary>HOW TO RUN CI</summary>
---

  By default, all the checks will be run automatically. Furthermore, when changing website-related stuff, the preview will be generated by the netlify bot.

  ### Heavy tests
  Add the [`heavy-tests`](/k8gb-io/k8gb/issues?q=is%3A*+label%3Aheavy-tests) label on this PR if you want full-blown tests that include more than 2-cluster scenarios.

  ### Debug tests
  If the test suite is failing for you, you may want to try triggering `Re-run all jobs` (top right) with [debug logging](https://docs.github.com/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging) enabled. It will also make the [print debug](/k8gb-io/k8gb/blob/master/.github/actions/print-debug/action.yaml) action more verbose.

</details>

### CODE REVIEW

**Comment by ytsarev** [comment]
@k0da CRD change, I would appreciate extra pair of eyes in review here, thanks :)

**Comment by k0da** [comment]
@kuritka do we really need to change CRD?

**Comment by kuritka** [comment]
@k0da , good question. Not IMO. 
Curretly  CRD is ochanged by controller-gen (in our case automatically when `make check` but can be disabled).

File: chart/k8gb/crd/k8gb.absa.oss_gslbs.yaml [hunk 1]
```diff
@@ -77,26 +83,31 @@ spec:
                         - kind
                         - name
                         type: object
+                        x-kubernetes-map-type: atomic
```
Comment (ytsarev) [inline chart/k8gb/crd/k8gb.absa.oss_gslbs.yaml:86]: SSA https://kubernetes.io/docs/reference/using-api/server-side-apply/

File: chart/k8gb/crd/k8gb.absa.oss_gslbs.yaml [hunk 2]
```diff
@@ -366,9 +372,3 @@ spec:
     storage: true
     subresources:
       status: {}
-status:
-  acceptedNames:
-    kind: ""
-    plural: ""
-  conditions: []
-  storedVersions: []
```
Comment (ytsarev) [inline chart/k8gb/crd/k8gb.absa.oss_gslbs.yaml:374]: looks like a fix for https://github.com/kubernetes-sigs/controller-tools/issues/456 , good to proceed 👍

File: chart/k8gb/crd/k8gb.absa.oss_gslbs.yaml (hunk 1 above)
Comment (ytsarev) [reply chart/k8gb/crd/k8gb.absa.oss_gslbs.yaml:86]:
```
Applicable to maps. atomic means that the map can only be entirely replaced by a single manager. granular means that the map supports separate managers updating individual fields.
```

Looks matching our case

**Review by ytsarev — APPROVED** [review]
I've double-checked the CRD changes, see inline comments. Looks safe to upgrade to me.

**Pull request merged**